drivers/include/digit7seg.h

```diff
@@ -0,0 +1,202 @@
+ *          a segment and dig* representing a digit pin to turn on.
+ * ```
+ *        --a--
+ *       |     |
+ *       f     b
+ *       |     |
+ *        --g--
+ *       |     |
+ *       e     c
+ *       |     |
+ *        --d--    .dp
+ *```
+ */
+typedef struct {
+    gpio_t data_a;  /**< Data input pin */
+    gpio_t data_b;  /**< Data input pin */
+    gpio_t data_c;  /**< Data input pin */
+    gpio_t data_d;  /**< Data input pin */
+    gpio_t data_e;  /**< Data input pin */
+    gpio_t data_f;  /**< Data input pin */
+    gpio_t data_g;  /**< Data input pin */
+    gpio_t data_dp; /**< Data input pin */
+    gpio_t dig1;    /**< Data input pin */
+    gpio_t dig2;    /**< Data input pin */
+    gpio_t dig3;    /**< Data input pin */
+    gpio_t dig4;    /**< Data input pin */
+    tim_t timer;    /**< Timer to use in digit7seg_power* */
+    uint8_t digits; /**< Number of digits */
+} digit7seg_params_t;
+
+/**
+ * @brief   Device descriptor for the DIGIT7SEG
+ */
+typedef struct {
+    digit7seg_params_t params; /**< Device parameters */
+    uint32_t value;            /**< Binary value to display */
+    uint8_t current_digit;     /**< Displayed digit */
+} digit7seg_t;
+
+/**
+ * @brief   Initialize the given DIGIT7SEG
+ *
+ * @param[out] dev      DIGIT7SEG device descriptor to initialize
+ * @param[in]  params   Device parameters to use
+ *
+ * @return  0 on success or negative error code, see #digit7seg_error_codes
+ */
+int digit7seg_init(digit7seg_t *dev, const digit7seg_params_t *params);
+
+/**
+ * @brief   Set the value for every digit
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] value     the value as an uint32_t
+ *
+ */
+void digit7seg_set_all_value(digit7seg_t *dev, uint32_t value);
+
+/**
+ * @brief   Set the value for one digit
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] index     the digit number between [0, DIGIT7SEG_MAX_DIGITS[
+ * @param[in] value     the value as an uint8_t, a is equal to the first bit
+ *                       b the second.... dp the 8th bit.
+ *
+ * @return              0 on success
+ * @return              -1 when an invalid index is passed
+ */
+int digit7seg_set_value(digit7seg_t *dev, int index, uint8_t value);
+
+#if MODULE_DIGIT7SEG_INT || DOXYGEN
+/**
+ * @brief   Set the value for every digit from an int
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] value     the value between [0, 9999]
+ *
+ */
+int digit7seg_set_int_all_value(digit7seg_t *dev, uint16_t value);
+
+/**
+ * @brief   Set the value for one digit from an int
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] index     the digit number between [0, DIGIT7SEG_MAX_DIGITS[
+ * @param[in] value     the value between [0, 9]
+ *
+ */
+int digit7seg_set_int_value(digit7seg_t *dev, int index, uint8_t value);
```

Comment:
```suggestion
int digit7seg_set_int_value(digit7seg_t *dev, unsigned index, uint8_t value);
```

Since negative indices are not allowed per API doc, we could encode this also into thr type.